### DISCUSSION TITLE
V extra bloat when generates C code

### DISCUSSION THREAD

**Discussion by Redysz** [vlang/v#19277] [General]
On the [compare](https://vlang.io/compare) page one can read:
> Unlike V, Nim generates unreadable C code with lots of extra bloat.

However, translating `println('Hello, World!')` to C with `v -o hello.c hello.v` command has much more bloat than nim. It produces 411.1kb c file ([here](https://pastebin.com/fRuXQmmt)) with many unreadable C parts, lie:
```c
Array_fixed_u64_309 _const_strconv__pos_exp = {((u64)(0x3ff0000000000000U)), ((u64)(0x4024000000000000U)), ((u64)(0x4059000000000000U)), ((u64)(0x408f400000000000U)), ((u64)(0x40c3880000000000U)), ((u64)(0x40f86a0000000000U)), ((u64)(0x412e848000000000U)), ((u64)(0x416312d000000000U)), ((u64)(0x4197d78400000000U)), ((u64)(0x41cdcd6500000000U)), ((u64)(0x4202a05f20000000U)), ((u64)(0x42374876e8000000U)), ((u64)(0x426d1a94a2000000U)), ((u64)(0x42a2309ce5400000U)), ((u64)(0x42d6bcc41e900000U)), ((u64)(0x430c6bf526340000U)), ((u64)(0x4341c37937e08000U)), ((u64)(0x4376345785d8a000U)), ((u64)(0x43abc16d674ec800U)), ((u64)(0x43e158e460913d00U)), ((u64)(0x4415af1d78b58c40U)), ((u64)(0x444b1ae4d6e2ef50U)), ((u64)(0x4480f0cf064dd592U)), ((u64)
[...]
```

For comparison sake: nim products 4.1kb c code which additionally include 18.7kb nimbase file.

will that be fixed? Can/Could/Will "translator" omit parts that are not used in project? What do you thing?

**Comment by JalonSolov** [comment]
Yes, of course it will be fixed.  V already supports the :construction: option `-skip-unused` which will reduce some of the output, and the compiler rewrite currently happening should make this much better in the future.  No, there is no projected date for when the work will be complete, but it has been started.
